build/config/BUILDCONFIG.gn

```diff
@@ -291,7 +291,12 @@ if (is_cobalt) {
   if (custom_toolchain != "") {
     cobalt_toolchain = custom_toolchain
   } else if (_default_toolchain != "") {
-    cobalt_toolchain = _default_toolchain
+    if (target_os == "linux" && is_clang) {
```

Comment:
This isn't what I was suggesting and I don't think it'll work—we need to ensure we're change the default toolchain, not just the `cobalt_toolchain` variable. Don't change this block at all, instead around line 281 above add a block

```
if (is_cobalt && target_os == "linux" && is_clang) {
  _toolchain_parts = string_split(_default_toolchain, ":")
  _toolchain_parts[1] = "cobalt_" + _toolchain_parts[1]
  _default_toolchain = string_join(":", _toolchain_parts)
}
```
Though ideally this could be done on the condition `is_cobalt` and not need to check the os/is_clang

Reply:
Do we need something like `current_toolchain_is_host_toolchain` that works regardless of the value for the `host_toolchain`?

Reply:
talking offline with andrew: it's a question of if we want 
 "//cobalt/build/toolchain/linux:cobalt_clang_$target_cpu"
or
"/build/toolchain/linux:cobalt_clang_x64"

i thoguht we prefer 1) option - lesser changes to upstream.

Reply:
to answer @y4vor's question : that can be accomplished with "current_toolchain == host_toolchain" .  
adding the variable seems unnecessary as the above conditional is already used in code: https://source.corp.google.com/search?q=%22current_toolchain%20%3D%3D%20host_toolchain%22&sq=&ss=h%2Fgithub%2Fyoutube%2Fcobalt%2F%2B%2Frefs%2Fheads%2Fmain

Reply:
Added according to andrew's suggestion .  This ended up creating a separation for all platforms.  That was a really good suggestion @andrewsavage1